# Week 1 anticipated schedule

This is the first week of the project. The primary targets in this week are:

1. Ensure all accounts are created (Github, Life Sciences)
2. Become familiar with using Git for source control. You should be able to create a repository, share it and use it to synchronise files
3. Begin learning Python - by the end of the week  you should be able to read in text from a file and manipulate it before writing out the results.
4. As a group identify candidate mitochondrial sequences for the first stage of analysis.

### Account registration
See the [Software](Software.md) page for details. You should be familiar with the working environment by the end of the week.

### Git for source control 
We will work through the exercises in the [Software Carpentry Git lesson](http://swcarpentry.github.io/git-novice/) as a group during this week. 

### Learning Python
The resources you use for learning Python are a matter of personal choice. Some links are included below.

#### Online
* [TryPython](http://trypython.org) - a good first dabble with an interactive web interpreter. You will need to move beyond this quite quickly
* [Python for Biologists](http://pythonforbiologists.com) - A course structured towards biologists. We have used this successfully with postgraduates in the past. You will need to register to get access to the data files.
* [Python reference](http://www.python.org) - The main documentation reference for Python. Get used to looking up details here.

#### Books
* [Computing Skills for Biologists](https://computingskillsforbiologists.com/) Very highly recommended.  

* [Practical Computing for Biologists](http://practicalcomputing.org/) - This book is a comprehensive reference that covers all you would need to know and some more. 
* [Computing for Biologists](http://www.cambridge.org/gb/academic/subjects/life-sciences/genomics-bioinformatics-and-systems-biology/computing-biologists-python-programming-and-principles) - Doesn't cover as much ground as the previous book but may be a more approachable style for some.
* [Python for Biologists](http://pythonforbiologists.com) - companion books for the Python for Biologists course.
* [Managing your Biological Data with Python](https://www.crcpress.com/Managing-Your-Biological-Data-with-Python/Via-Rother-Tramontano/9781439880937) - Takes a story based approach where each chapter solves a problem and introduces programming concepts along the way.

(Conflicts of interest - I know the authors of the latter two books) I have copies of each of these which can be inspected on request.
There are also many free PDF books available that introduce Python as a programming language for non/novice-programmers

## Lab meeting
We will have a lab meeting at the first available opportunity during this week. I will email you all during the previous week to identify a suitable time.
 All lab meetings will be held in Carnelley Museum.
The agenda for the lab meeting is provisionally;

* Introduction and welcome
* The practical - biological overview
* The practical - technical overview
* Getting started - Introducing the software
* Performance Expectations - lab behaviour, record keeping etc. 
* Dates and times - Git tutorial session. Future lab meetings and roles.

There will be at least one formal lab meeting during the week, and no more than two. This should (given the anticipated numbers) allow everyone a chance to take on the roles of presenter, chair, and minute taker.

## Mitochondrial sequences
You should use the [NCBI interface to the nucleotide database](http://www.ncbi.nlm.nih.gov/nuccore) to identify complete mitochondrial genomes for appropriate species. Think about which species you may wish to consider (wild mammals found in Scotland). As you select species ensure you record the full scientific name, any sequence accession numbers and some notes about the species. This should be done as a collaborative effort. With the ability to work together using Git, you can build up a series of small pages, with links, that give a summary of each species chosen. 